app/src/main/java/fr/free/nrw/commons/nearby/fragments/NearbyParentFragment.java

```diff
@@ -1238,7 +1184,7 @@ public void enableFABRecenter() {
     public void addCurrentLocationMarker(final fr.free.nrw.commons.location.LatLng curLatLng) {
         if (null != curLatLng && !isPermissionDenied) {
             ExecutorUtils.get().submit(() -> {
-                mapView.post(() -> removeCurrentLocationMarker());
```

Comment:
Moving these to the off main thread to avoid blocking the main thread. Found from the logs -  "Too much work on main thread"